Sources/Sentry/SentryEnvelopeRateLimit.m

```diff
@@ -0,0 +1,67 @@
+#import <Foundation/Foundation.h>
+#import "SentryEnvelopeRateLimit.h"
+#import "SentryRateLimits.h"
+#import "SentryEnvelope.h"
+#import "SentryRateLimitCategoryMapper.h"
+
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SentryEnvelopeRateLimit ()
+
+@property(nonatomic, strong) id <SentryRateLimits> rateLimits;
+
+@end
+
+@implementation SentryEnvelopeRateLimit
+
+- (instancetype) initWithRateLimits:(id<SentryRateLimits>) sentryRateLimits {
+    if (self = [super init]) {
+        self.rateLimits = sentryRateLimits;
+    }
+    return self;
+}
+
+- (SentryEnvelope *)removeRateLimitedItems:(SentryEnvelope *)envelope {
+    SentryEnvelope *result = envelope;
+    
+    NSArray<SentryEnvelopeItem *> *itemsToDrop = [self getEnvelopeItemsToDrop:envelope.items];
+    
+    if (itemsToDrop.count > 0) {
+        NSArray<SentryEnvelopeItem *> *itemsToSend = [self getItemsToSend:envelope.items withItemsToDrop:itemsToDrop];
+        
+        result = [[SentryEnvelope alloc] initWithId:envelope.header.eventId items:itemsToSend];
```

Comment:
m: I'm afraid this code won't version well. I understand it's what the current API offers us to create a new envelope but if we add new fields to `SentryEnvelope` or new header values, they will be lost during the conversion.

Ideally we'd create a new constructor so we can call this like:

```
         result = [[SentryEnvelope alloc] initWithHeader:envelope.header items:itemsToSend];
```

At least this way we pass the original header down to the new instance.

Reply:
I'm going to fix this. This should be high priority IMO.

Reply:
Done: https://github.com/getsentry/sentry-cocoa/pull/478